import hymn:error (exception)
import hymn:result (result)

def okie() result<int,exception>
    return result.ok(1)

def not_okie() result<int,exception>
    return result.error(exception("1", "dang it!"))

def compile() result<string,exception>
    ex = try okie() catch(e) return result.error(e.value)
    return result.ok(str(ex))

def compile_2() result<string,exception>
    ex = try not_okie() catch(e)
        return result.error(e.value)
    return result.ok(str(ex))

def main
    ok = compile()
    match ok: ok(r): echo("ok:", r.value)

    match compile_2(): error(e)
        echo("error:", e.value.get_description())

